### DISCUSSION TITLE
How to call pdf-converter.py

### DISCUSSION THREAD

**Discussion by goldendel** [pymupdf/PyMuPDF#1079] [Looking for help] [answered]
Hello.

I am new to python. I am trying to call pdf-converter.py in the following for loop instead of printing the file name when it finds an .oxps file:

```
`from pathlib import Path
import os
import pdf_converter
pdfconverter = 'C:/Users/timth/OneDrive/Desktop/pdf_converter.py'

for path in Path('C:\Apps\Test').rglob('*.oxps'):
    print(path.name)
    #exec(open(pdfconverter).read(path.name))
    #pdfconverter = ('C:/Users/timth/OneDrive/Desktop/pdf_converter.py' path.name)
    #os.system(pdfconverter)
    #filetoconvert = path.name
    #pdf-converter,filetoconvert`
```

However, as pdf-converter.py requires the filename to convert as a parameter, I can't use import, or I get the following error:

```
`File "C:\Users\timth\OneDrive\Desktop\pdf_converter.py", line 24, in <module>
    fn = sys.argv[1]
IndexError: list index out of range`
```

I have tried 

`exec(open('file.py').read())`
and
`os.system('python file.py')`
as a last resort.  `exec(open('file.py').read())` doesn't work and `os.system('python file.py')` flashes up the file and fails, I assume becuase of a hidden error.  What is the correct way to call pdf-converter.py with the filename is has found in the loop?

Thanks for your help and advice.

Tim

**Comment by JorjMcKie** [comment, marked as answer]
This is an example script intended to be invoked standalone like as usual: ``python pdf-converter.py some-file-to-convert``.
It is **_not intended_** to be imported ...
All the scripts in this repo mainly are meant to be used as templates or starting point to develop an own solution - not necessarily as stable, fool-proof utilities, guaranteed to always work, etc.

It is of course possible to convert this script to an importable module - no major problem, but some work is still required.

**Reply by goldendel** [reply]
OK Thanks @JorjMcKie.  For the minute, I'll write a loop in a language I'm more used to and call your script as standalone.  I haven't written an importable module as yet.

Thanks for your advice.

Tim
